include/seqan3/io/structure_file/format_vienna.hpp

```diff
@@ -216,42 +216,46 @@ class structure_file_input_format_REMOVEME<format_vienna>
         }
 
         // READ STRUCTURE (if present)
+        [[maybe_unused]] size_t structure_length{};
         if constexpr (!detail::decays_to_ignore_v<structure_type>)
         {
             if constexpr (structured_seq_combined)
             {
                 assert(std::addressof(seq) == std::addressof(structure));
                 using alph_type = typename value_type_t<structure_type>::structure_alphabet_type;
-                std::ranges::copy(read_structure<alph_type>(stream_view), begin(structure));
+                // We need the structure_length parameter to count the length of the structure while reading
+                // because we cannot infer it from the (already resized) structure_seq object.
+                std::ranges::copy(read_structure<alph_type>(stream_view, structure_length), begin(structure));
 
                 if constexpr (!detail::decays_to_ignore_v<bpp_type>)
                     detail::bpp_from_rna_structure<alph_type>(bpp, structure);
             }
             else
             {
                 using alph_type = value_type_t<structure_type>;
-                std::ranges::copy(read_structure<alph_type>(stream_view), std::back_inserter(structure));
+                std::ranges::copy(read_structure<alph_type>(stream_view, structure_length),
```

Comment:
Just wondering. What do we know about the type of `structure`?  I see size was used on it, so we assume sized_range? It is also likely that we parse the input always in a vector or another object with probably having a sized_sentinel? 
If all of this is true then we can get the length of the structure quite easily. 
Either it is std::ranges::distance/size whatever on the structure that is stand alone or 
`std::ranges::distance(std::ranges::begin(structure), copy_result.second)` if it is part of the sequence.
Not exactly sure what the second member of copy_result is called.

Reply:
Currently we do not enforce the range to be a sized range (The default is std::vector though).
So std::distance would be linear again in case it is not sized. I don't know if we want to be restricted to that because we usually just want to fill the container with values?

Reply:
But that's the whole point. If we argue that we don't care if you use a non performant data type, so be it. But the default is always sized. Then we can as well use the information we already get returned from the copy algorithm instead of injecting it into another linear one. that means, if std::distance is already slow on the user provided type and we use it anyway, it doesn't matter if it is called one or two times. As a matter of fact, right now, it will always do some additional linear operation. So we didn't gain anything.